indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,534 @@
+                        resultsWereOpen = node.closest('.typeahead-container').hasClass('result');
+                    },
+                    onResult: function(node, query, result, resultCount, resultCountPerGroup) {
+                        node.parent().find('.just-use-option').toggleClass('active', resultCount === 0);
+                        justUseOptionSelected = resultCount === 0;
+                    }
+
+                }
+            });
+
+            roomInput.typeahead({
+                source: rooms,
+                minLength: 0,
+                searchOnFocus: true,
+                group: true,
+                maxItem: 0,
+                resultContainer: '#' + roomInput.siblings('.typeahead-results').find('.results-list-container').attr('id'),
+                template: {% raw %}'<span id="location-room-{{id}}">{{name}}</span>'{% endraw %},
+                emptyTemplate: function(query) { return ""; },
+                display: 'name',
+                callback: {
+                    onInit: function(node) {
+                        this.query = node.val();
+                    },
+                    onClickBefore: function(node, a, item, event) {
+                        validRoom = true;
+                    },
+                    onClickAfter: function(node, a, item, event) {
+                        updateVenueOnRoomChange(item);
+
+                        hiddenData.room_id = item.id;
+                        delete hiddenData.room_name;
+
+                        postSelectionActions(roomInput, item.name);
+                        latestSelectedOption = {field: node.attr('id'), value: item.name}
+                        highlightOption(roomInput, item.name);
+                    },
+                    onLayoutBuiltBefore: function(node, query, result, resultHtmlList) {
+                        return updateResultsList(node, resultHtmlList);
+                    },
+                    onSearch: function(node, query) {
+                        this.query = cleanQuery(query);
+                    },
+                    onShowLayout: function(node, query) {
+                        roomInput.next('.highlighter').removeClass('visible');
+                        currentScrollPos = 0;
+                    },
+                    onHideLayout: function(node, query) {
+                        if (justUseOptionSelected) {
+                            roomInput.trigger('typeahead:click-on-custom-option', {item_name: query});
+                        } else if (!validRoom && node.val() && !preventGuessing) {
+                            var result = [];
+                            var venueValue = venueInput.val();
+                            if (venueValue) {
+                                result = findRoomInVenue(query, venueValue);
+                            } else {
+                                var count = 0;
+                                while (count < venueNames.length && result.length == 0) {
```

Comment:
why not a `for` loop? `for (var i = 0; i < venueNames.length && !result.length; i++)`